src/tools/illink/Directory.Build.props

```diff
@@ -1,8 +1,4 @@
 <Project>
-  <PropertyGroup>
-    <!-- Don't use the prebuilt illink that is referenced by Arcade. -->
-    <UsingToolMicrosoftNetILLinkTasks>false</UsingToolMicrosoftNetILLinkTasks>
-  </PropertyGroup>
```

Comment:
This is the root cause of source-build failure. It is causing the usage of prebuilt version of Microsoft.Net.ILLink.Tasks package. Is this accidental? @akoeplinger @vikoeplinger

Reply:
I suspect this would would in the product source-build because the ILLink version we have in n-1 artifacts would be coherent with the SDK we use to build therefore it wouldn't be a prebuilt.  During the repo leg this may need to be marked as an allowed prebuilt.  This full product source build should be validated.

Reply:
> I suspect this would would in the product source-build because the ILLink version we have in n-1 artifacts would be coherent with the SDK we use to build therefore it wouldn't be a prebuilt. During the repo leg this may need to be marked as an allowed prebuilt. This full product source build should be validated.

I'll validate full product build locally.

Reply:
> > I suspect this would would in the product source-build because the ILLink version we have in n-1 artifacts would be coherent with the SDK we use to build therefore it wouldn't be a prebuilt. During the repo leg this may need to be marked as an allowed prebuilt. This full product source build should be validated.
> 
> I'll validate full product build locally.

Full source-build is showing 20 prebuilts but not the one for `Microsoft.Net.ILLink.Tasks`. I'm unsure about this - @MichaelSimons 

Here's the full list of prebuilts:
```
    <Usage Id="Microsoft.CodeAnalysis.Analyzers" Version="3.3.4" File="src/runtime/artifacts/source-build/self/src/artifacts/obj/Microsoft.Extensions.Configuration.Binder.SourceGeneration/project.assets.json" />
    <Usage Id="Microsoft.CodeAnalysis.Common" Version="4.7.0-3.23314.3" File="src/runtime/artifacts/source-build/self/src/artifacts/obj/Microsoft.Extensions.Configuration.Binder.SourceGeneration/project.assets.json" />
    <Usage Id="Microsoft.CodeAnalysis.CSharp" Version="4.7.0-3.23314.3" File="src/runtime/artifacts/source-build/self/src/artifacts/obj/Microsoft.Extensions.Configuration.Binder.SourceGeneration/project.assets.json" />
    <Usage Id="Microsoft.CodeAnalysis.CSharp.Workspaces" Version="4.7.0-3.23314.3" File="src/runtime/artifacts/source-build/self/src/artifacts/obj/Microsoft.Extensions.Configuration.Binder.SourceGeneration/project.assets.json" IsDirectDependency="true" />
    <Usage Id="Microsoft.CodeAnalysis.Workspaces.Common" Version="4.7.0-3.23314.3" File="src/runtime/artifacts/source-build/self/src/artifacts/obj/Microsoft.Extensions.Configuration.Binder.SourceGeneration/project.assets.json" />
    <Usage Id="Microsoft.Extensions.DependencyInjection" Version="8.0.0-rc.1.23414.4" File="src/sdk/artifacts/source-build/self/src/artifacts/obj/Microsoft.NET.Build.Containers/project.assets.json" />
    <Usage Id="Microsoft.Extensions.DependencyInjection.Abstractions" Version="8.0.0-rc.1.23414.4" File="src/sdk/artifacts/source-build/self/src/artifacts/obj/Microsoft.NET.Build.Containers/project.assets.json" />
    <Usage Id="Microsoft.Extensions.Logging" Version="8.0.0-rc.1.23414.4" File="src/sdk/artifacts/source-build/self/src/artifacts/obj/Microsoft.NET.Build.Containers/project.assets.json" IsDirectDependency="true" />
    <Usage Id="Microsoft.Extensions.Logging.Abstractions" Version="8.0.0-rc.1.23414.4" File="src/sdk/artifacts/source-build/self/src/artifacts/obj/Microsoft.NET.Build.Containers/project.assets.json" IsDirectDependency="true" />
    <Usage Id="Microsoft.Extensions.Options" Version="8.0.0-rc.1.23414.4" File="src/sdk/artifacts/source-build/self/src/artifacts/obj/Microsoft.NET.Build.Containers/project.assets.json" />
    <Usage Id="Microsoft.Extensions.Primitives" Version="8.0.0-rc.1.23414.4" File="src/sdk/artifacts/source-build/self/src/artifacts/obj/Microsoft.NET.Build.Containers/project.assets.json" />
    <Usage Id="Microsoft.NET.Workload.Mono.ToolChain.Current.Manifest-8.0.100-rc.1" Version="8.0.0-rc.1.23414.4" />
    <Usage Id="Microsoft.NET.Workload.Mono.ToolChain.net6.Manifest-8.0.100-rc.1" Version="8.0.0-rc.1.23414.4" />
    <Usage Id="Microsoft.NET.Workload.Mono.ToolChain.net7.Manifest-8.0.100-rc.1" Version="8.0.0-rc.1.23414.4" />
    <Usage Id="System.Composition" Version="7.0.0" File="src/runtime/artifacts/source-build/self/src/artifacts/obj/Microsoft.Extensions.Configuration.Binder.SourceGeneration/project.assets.json" />
    <Usage Id="System.Composition.AttributedModel" Version="7.0.0" File="src/runtime/artifacts/source-build/self/src/artifacts/obj/Microsoft.Extensions.Configuration.Binder.SourceGeneration/project.assets.json" />
    <Usage Id="System.Composition.Convention" Version="7.0.0" File="src/runtime/artifacts/source-build/self/src/artifacts/obj/Microsoft.Extensions.Configuration.Binder.SourceGeneration/project.assets.json" />
    <Usage Id="System.Composition.Hosting" Version="7.0.0" File="src/runtime/artifacts/source-build/self/src/artifacts/obj/Microsoft.Extensions.Configuration.Binder.SourceGeneration/project.assets.json" />
    <Usage Id="System.Composition.Runtime" Version="7.0.0" File="src/runtime/artifacts/source-build/self/src/artifacts/obj/Microsoft.Extensions.Configuration.Binder.SourceGeneration/project.assets.json" />
    <Usage Id="System.Composition.TypedParts" Version="7.0.0" File="src/runtime/artifacts/source-build/self/src/artifacts/obj/Microsoft.Extensions.Configuration.Binder.SourceGeneration/project.assets.json" />
```

Reply:
There is only one direct dependency coming from runtime

    <Usage Id="Microsoft.CodeAnalysis.CSharp.Workspaces" Version="4.7.0-3.23314.3" File="src/runtime/artifacts/source-build/self/src/artifacts/obj/Microsoft.Extensions.Configuration.Binder.SourceGeneration/project.assets.json" IsDirectDependency="true" />

This is a result of https://github.com/dotnet/runtime/pull/90340 which was addressed in https://github.com/dotnet/runtime/pull/91006.  I suspect the commit tested didn't have the fix.  I think the test has proven this will not introduce a prebuilt in the product level source-build.

Reply:
> There is only one direct dependency coming from runtime
> 
> ```
> <Usage Id="Microsoft.CodeAnalysis.CSharp.Workspaces" Version="4.7.0-3.23314.3" File="src/runtime/artifacts/source-build/self/src/artifacts/obj/Microsoft.Extensions.Configuration.Binder.SourceGeneration/project.assets.json" IsDirectDependency="true" />
> ```
> 
> This is a result of #90340 which was addressed in #91006. I suspect the commit tested didn't have the fix. I think the test has proven this will not introduce a prebuilt in the product level source-build.

Thanks - I'll work on a fix to add exclusion for ILLink.Tasks.